src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java

```diff
@@ -0,0 +1,219 @@
+      endTimeTracker = "" + getCurrentTime();
+    } else {
+      endTimeTracker = "" + (startTimeTrackerLong + getQueueFifoTimeSpanMillis());
+    }
+    final long endTimeTrackerLong = Long.parseLong(endTimeTracker);
+
+    // Run EBD export call
+    final String ebdFileName =
+        HistoricalDataConstants.QUEUE_FILE_FOLDER
+            + HistoricalDataConstants.QUEUE_EBD_FILE_NAME
+            + HistoricalDataConstants.QUEUE_FILE_EXTENSION;
+    final String ebdStartTime = convertToEBDTimeFormat(startTimeTrackerLong);
+    final String ebdEndTime = convertToEBDTimeFormat(endTimeTrackerLong);
+    HistoricalDataManager.exportHistoricalToFile(
+        ebdStartTime,
+        ebdEndTime,
+        ebdFileName,
+        includeTagGroupA,
+        includeTagGroupB,
+        includeTagGroupC,
+        includeTagGroupD);
+
+    // Parse EBD export call
+    ArrayList queueData = HistoricalDataManager.parseHistoricalFile(ebdFileName);
+
+    // Store end time +1 ms (to prevent duplicate data)
+    final String newTimeTrackerVal = "" + (endTimeTrackerLong + 1);
+    writeDataFile(timeMarkerFileName, newTimeTrackerVal);
+
+    // Return data
+    return queueData;
+  }
+
+  /**
+   * Reads the specified file into a string.
+   *
+   * @param filename file to read
+   * @return file contents
+   * @throws IOException if unable to access or read file
+   */
+  private static String readDataFile(String filename) throws IOException {
+    // Create input stream and buffered reader
+    InputStream inputStream = new FileInputStream(filename);
+    BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+
+    // Create string for file contents
+    StringBuffer fileContents = new StringBuffer();
+
+    // Loop through file and add to string
+    String currentLine = bufferedReader.readLine();
+    while (currentLine != null) {
+      // Append current line
+      fileContents.append(currentLine);
+
+      // Read next line
+      currentLine = bufferedReader.readLine();
+
+      // If next line exists, add new line to string
+      fileContents.append("\n");
+    }
+
+    // Close reader and stream
+    bufferedReader.close();
+    inputStream.close();
+
+    return fileContents.toString();
+  }
+
+  /**
+   * Writes the specified string to the specified file
+   *
+   * @param filename file to write to
+   * @param contents file contents
+   * @throws IOException if unable to access or write file
+   */
+  private static void writeDataFile(String filename, String contents) throws IOException {
+    // Create output stream and buffered writer
+    OutputStream outputStream = new FileOutputStream(filename);
+    BufferedWriter bufferedWriter = new BufferedWriter(new OutputStreamWriter(outputStream));
+
+    // Write contents to file
+    bufferedWriter.write(contents);
+
+    // Close writer and stream
+    bufferedWriter.close();
+    outputStream.close();
+  }
+}
```

Comment:
should these be moved to their own class or even library? They are pretty generic.

Reply:
I can move these to a library. Do you think these belong in any of the existing libraries, or should I do a new one?

Reply:
I dont think they quite fit in the string lib. Maybe sc-flexy-file-tools

Reply:
moved